### ISSUE TITLE
`eleventyConfig.setDataFileSuffixes([""])` not working

### ISSUE THREAD

**Issue by patrulea** [11ty/eleventy#3066]
### Operating system

macOS Sonoma 14.0

### Eleventy

2.0.1

### Describe the bug

I’m trying to use files named `_data.js` and `_data.json` as front-matter defaults for templates on a per-directory basis.

Though, using both `setDataFileBaseName` and `setDataFileSuffixes` settings in combination won’t let me achieve that, as the data files are completely ignored.

### Reproduction steps

```js
module.exports = (eleventyConfig) => {
  eleventyConfig.setDataFileBaseName("_data");
  eleventyConfig.setDataFileSuffixes([""]);
  return {
    jsDataFileSuffix: ""
  };
};
```

### Expected behavior

Data stored in directory files named `_data.js` or `_data.json` should be treated as front-matter for sibling template files.

**Comment by patrulea** [comment]
I made it work the way I wanted but it feels hacky.
```js
module.exports = (eleventyConfig) => {
  eleventyConfig.setDataFileBaseName("_");
  eleventyConfig.setDataFileSuffixes(["data"]);
};
```
I found out what was causing a problem was `setDataFileSuffixes([""])`. For some reason, I’m not being able to set an empty suffix for directory data files.

I tried passing an array containing an empty string in combination with another value.
```js
setDataFileSuffixes([
  "",
  "data"
]
```
The empty string would be “ignored” and only files suffixed with “data” would work.
